doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+# User Management design #
+
+##  1. Table of Content
+
+
+* 1. [Table of Content](#TableofContent)
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+		* 1.5.1. [Define defaults](#Definedefaults)
+		* 1.5.2. [Configurations](#Configurations)
+* 2. [Design](#Design)
+	* 2.1. [Architecture Design](#ArchitectureDesign)
+	* 2.2. [High-Level Design](#High-LevelDesign)
+	* 2.2. [Usercfd Role](#UsercfdRole)
+* 3. [Configuration and management](#Configurationandmanagement)
+		* 3.1. [ConfigDB Tables](#ConfigDBTables)
+		* 3.2. [ConfigDB schemas](#ConfigDBschemas)
+		* 3.3. [CLI/YANG model](#CLIYANGmodel)
+* 4. [Build](#Build)
+		* 4.1. [Compilation](#Compilation)
```

Comment:
suggest to have runtime configuration only, not build time. enablement can be controlled by config db, it would be easier for user to consume this feature.

Reply:
+1 -- can you also share the specific reasons why this is starting off as disabled by default? Are we concerned that this design may not be accepted by all users -- if so, can we update the design in such a way that it is?

Reply:
We have received a request from several vendors to add a compilation flag as they are using remote authentication only.
If we change it to be controlled in runtime, it will necessitate usercfd to run always, which will undoubtedly have an impact on the initialization time even when the feature is disabled.